### PR TITLE
#36422 - Add footer button scrollToTop

### LINKED ISSUE #218: Add scroll to top footer button

### PR DESCRIPTION
**Pull request by kjugi**
+ Add script with scroll to top
+ Modify variables and paddings
+ Add config to button
+ Add button in module preview

Fixes #218 

https://dev.snowdog.pro/issues/36422

### CODE REVIEW

File: components/03-modules/footer/_footer.scss [hunk 1]
```diff
@@ -133,4 +134,23 @@ $footer__copywrite-padding-left\@large : $spacer--medium;
             padding-left: $footer__copywrite-padding-left\@large;
```
Comment (Januszpl) [inline components/03-modules/footer/_footer.scss]: shouldn't be `\@medium` ?

File: components/03-modules/footer/_footer.scss [hunk 2]
```diff
@@ -133,4 +134,23 @@ $footer__copywrite-padding-left\@large : $spacer--medium;
...
+        &:focus {
+            .footer__icon {
+                fill: $footer__icon-fill-hover;
+            }
+        }
+    }
+
+    &__icon {
+        width: $footer__icon-sizing;
+        height: $footer__icon-sizing;
```
Comment (Januszpl) [inline components/03-modules/footer/_footer.scss]: icon-size seems better name ;) anyway what if in some project height would be different? I also used sometimes just "-size" but not sure if we shouldn't provide vars for width and height

Comment (kjugi) [reply components/03-modules/footer/_footer.scss]: So far when you have child components based on alpaca when you want to update styles we are overwriting the file and we can change the variables and all other stuff here. Not necessary to setting 2 different variables

Comment (kjugi) [reply components/03-modules/footer/_footer.scss]: and renamed to `size`

File: components/03-modules/footer/_footer.scss (hunk 1 above)
Comment (kjugi) [reply components/03-modules/footer/_footer.scss]: nope

Comment (Januszpl) [reply components/03-modules/footer/_footer.scss]: ok it's different line. it's ok

**Pull request merged**